socs/Lakeshore/Lakeshore336.py

```diff
@@ -0,0 +1,1195 @@
+        :params D: derivative term in PID loop
+        :type D: float
+        :returns: response from PID command
+        :rtype: str
+        """
+        assert float(P) <= 1000 and float(P) >= 0.1
+        assert float(I) <= 1000 and float(I) >= 0.1
+        assert float(D) <= 200 and float(D) >= 0
+
+        self.P = P
+        self.I = I
+        self.D = D
+
+        resp = self.ls.msg(f"PID {self.output},{P},{I},{D}")
+        return resp
+
+    def get_ramp(self):
+        '''Return list of params <on/off>, <rate value> for RAMP? msg
+
+        Returns
+        -------
+        list
+            <on/off>, <rate value> for RAMP? msg
+        '''
+        resp = self.ls.msg(f'RAMP? {self.output}').split(',')
+        self.ramp_enabled = ramp_key[resp[0]]
+        self.ramp_rate = float(resp[1])
+        return resp
+
+    def _set_ramp(self, params):
+        '''Set the RAMP params <on/off>, <rate value>
+
+        Parameters
+        ----------
+        params : list
+            <on/off>, <rate value>
+        '''
+        assert len(params) == 2
+
+        reply = [str(self.output)]
+        [reply.append(x) for x in params]
+
+        param_str = ','.join(reply)
+        resp = self.ls.msg(f'RAMP {param_str}')
+        return resp
+
+    def get_ramp_on_off(self):
+        '''Get string indicating ramp 'on' or 'off' '''
+        self.get_ramp()
+        return self.ramp_enabled
+
+    def set_ramp_on_off(self, on_off):
+        '''Turn ramp on or off
+
+        Parameters
+        ----------
+        on_off : str
+            Either 'on' to enable ramp or 'off' to disable ramp
+        '''
+        assert on_off.lower() in ramp_lock
+
+        resp = self.get_ramp()
+        resp[0] = ramp_lock[on_off.lower()]
+        self.ramp_enabled = on_off.lower()
+        return self._set_ramp(resp)
+
+    def get_ramp_rate(self):
+        '''Returns ramp rate in K/min'''
+        self.get_ramp()
+        return self.ramp_rate
+
+    def set_ramp_rate(self, rate):
+        '''Set ramp rate of changes in setpoint, in K/min
+
+        Parameters
+        ----------
+        rate : int or float
+            Absolute value of setpoint rate of change, from 0.1 to 100 K/min
+        '''
+        assert rate >= 0.1 and rate <= 100
+
+        resp = self.get_ramp()
+        resp[1] = str(rate)
+        self.ramp_rate = rate
+        return self._set_ramp(resp)
+
+    def get_ramp_status(self):
+        '''Return a string indicating whether or not the setpoint
+        is currently ramping'''
+        resp = self.ls.msg(f'RAMPST? {self.output}')
+        ramp_stat_dict = {'0': 'Not ramping', '1': 'Ramping'}
+        self.ramp_status = ramp_stat_dict[resp]
+        return self.ramp_status
+
+    def get_heater_percent(self):
+        '''Return the current heater output level in %'''
+        resp = self.ls.msg(f'HTR? {self.output}')
+        self.percent = float(resp)
+        return self.percent
+
+# Do stuff
+
+if __name__ == '__main__':
```

Comment:
I find it most useful to give the option here to connect to any device given an argument. That allows one to run this file interactively and perform some actions with a Lakeshore device, so like in the 372 it initializes `ls = LS372(sys.argv[1])`, which allows me to run something like `python3 -i Lakeshore372.py 192.168.1.2` and get the 372 at that address, and interact as needed.

I'd recommend something similar here for consistency with the other Lakeshore drivers, which will include using `ls` instead of `box` and leaving the comm open.

Reply:
done